src/scribe_data/check/check_language_metadata.py

```diff
@@ -0,0 +1,231 @@
+        else:
+            # Validate the parent language itself
+            if "qid" not in details:
+                missing_qids.append(lang)
+            if "iso" not in details:
+                missing_isos.append(lang)
+
+    return {"missing_qids": missing_qids, "missing_isos": missing_isos}
+
+
+def check_language_metadata():
+    """
+    Validates language metadata by performing the following checks:
+
+    1. Ensures that all languages listed in `language_data_extraction` are present in `language_metadata.json`, and vice versa.
+    2. Checks if each language in `language_metadata.json` has the required properties:
+       - 'qid' (a unique identifier)
+       - 'iso' (ISO language code)
+
+    This function helps identify missing languages or missing properties, ensuring data consistency across both sources.
+    """
+    languages_in_directory = get_available_languages()
+    missing_languages_metadata = get_missing_languages(
+        languages_in_metadata, languages_in_directory
+    )
+    missing_languages_extraction = get_missing_languages(
+        languages_in_directory, languages_in_metadata
+    )
+    languages_with_missing_properties = validate_language_properties(
+        languages_in_metadata
+    )
+
+    if (
+        missing_languages_metadata
+        or missing_languages_extraction
+        or languages_with_missing_properties["missing_qids"]
+        or languages_with_missing_properties["missing_isos"]
+    ):
+        if missing_languages_metadata or missing_languages_extraction:
+            print(
+                "There are missing languages or inconsistencies between language_metadata.json and language_data_extraction.\n"
+            )
+
+        if missing_languages_metadata:
+            print("Languages missing from language_metadata.json:")
+            for lang in missing_languages_metadata:
+                print(f"  • {lang.title()}")
+
+        if missing_languages_extraction:
+            print("\nLanguages missing from language_data_extraction:")
+            for lang in missing_languages_extraction:
+                print(f"  • {lang.title()}")
+
+        if languages_with_missing_properties["missing_qids"]:
+            print("\nLanguages missing the `qid` property:")
+            for lang in languages_with_missing_properties["missing_qids"]:
+                print(f"  • {lang.title()}")
+
+        if languages_with_missing_properties["missing_isos"]:
+            print("\nLanguages missing the `iso` property:")
+            for lang in languages_with_missing_properties["missing_isos"]:
+                print(f"  • {lang.title()}")
+
+        # Exit with a non-zero status code to indicate failure
+        sys.exit(1)  # Indicate failure
+
+    print(
+        "All languages match between language_metadata.json and language_data_extraction; languages in language_metadata.json have the correct properties."
+    )
+
+
+if __name__ == "__main__":
+    check_language_metadata()
```

Comment:
If this is for a test, then it's fine. but we will be calling this in  `check_project_metadata.yaml` file so no need for this :)

Reply:
thank you! good to know 😊